### DISCUSSION TITLE
Incorrectly starts a codespace on `main` when repo is on a tag/branch

### DISCUSSION THREAD

**Discussion by josephrocca** [community/community#43548] [Codespaces]
### Select Topic Area

Bug

### Body

Steps to reproduce:

1. Go to this URL: https://github.com/onnx/onnx/tree/v1.13.0
2. Create a new codespace using the green button
3. The new codespace tab will open
4. Go back to the previous tab and open *another* codespace
5. Notice that it opens on `main` instead of v1.13.0

Video of the above steps:

https://user-images.githubusercontent.com/1167575/211135358-986b81a8-8ff7-4546-ab9c-7bdc13cab559.mp4

**Comment by aditya30394** [comment]
Thanks for bringing this to our notice. I will let the relevant team know about this.
